odoo/addons/base/models/ir_module.py

```diff
@@ -689,6 +689,15 @@ def button_upgrade(self):
                 if dep.state == 'uninstalled':
                     to_install += self.search([('name', '=', dep.name)]).ids
 
+        if 'base' in self.mapped('name'):
```

Comment:
What about moving that `if` statement above?  Your code looks a bit like: "follow the complex path first, then realize you could have taken a shortcut."  Selecting all installed modules is clearly an optimization of the `while` loop above, isn't it?

I also suggest to target version 13.0 (our oldest maintained version), since this issue is not specific to version 14.0.

Reply:
OK, I hope this is what you mean: https://github.com/odoo/odoo/pull/85516